leshan-lwm2m-core/src/main/java/org/eclipse/leshan/core/response/ReadCompositeResponse.java

```diff
@@ -16,19 +16,58 @@
 package org.eclipse.leshan.core.response;
 
 import java.util.Map;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.stream.Collectors;
 
 import org.eclipse.leshan.core.ResponseCode;
 import org.eclipse.leshan.core.node.LwM2mNode;
 import org.eclipse.leshan.core.node.LwM2mPath;
+import org.eclipse.leshan.core.node.TimestampedLwM2mNodes;
 
 public class ReadCompositeResponse extends AbstractLwM2mResponse {
 
     protected final Map<LwM2mPath, LwM2mNode> content;
 
-    public ReadCompositeResponse(ResponseCode code, Map<LwM2mPath, LwM2mNode> content, String errorMessage,
-            Object coapResponse) {
+    protected final TimestampedLwM2mNodes timestampedValues;
+
+    public ReadCompositeResponse(ResponseCode code, Map<LwM2mPath, LwM2mNode> content,
+            TimestampedLwM2mNodes timestampedValues, String errorMessage, Object coapResponse) {
         super(code, errorMessage, coapResponse);
-        this.content = content;
+
+        Map<LwM2mPath, LwM2mNode> responseContent;
+        TimestampedLwM2mNodes responsetimestampedValues;
+
+        if (timestampedValues != null) {
+            // handle if timestamped value is passed
+            if (content != null) {
+                throw new IllegalArgumentException("content OR timestampedValue should be passed but not both");
+            }
```

Comment:
I think you should ensure there is only 1 timestamp here.
with something like : `timestampedValues.getTimestamps().size() > 1`

I don't know if size == 0 make senses and what we want to do in that case.

If `timestampedValues.getTimestamps().size() == 1`  we can store the map timestamp in variable this will make following code easier to read.

```java
// get only timestamp available. 
Instant timestamp = timestampedValues.getTimestamps().iterator().next();
```

Reply:
OK - corrected